File: src/main/java/ru/bebriki/bebriki/models/Good.java
package ru.bebriki.bebriki.models;

import jakarta.persistence.*;
import lombok.*;

import java.util.List;
import java.util.Set;

@Builder
@Data
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Entity
@Table(name = "goods")
public class Good {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    @Column(name = "title")
    private String title;
    @Column(name = "category")
    private String category;
    @Column(name = "price")
    private Integer price;
    @Column(name = "discount")
    private Integer discount;
    @Column(name = "description")
    private String description;
    @Column(name = "imageURL")
    private String imageURL;
    @Column(name = "amount")
    private Integer amount;
//    @ManyToMany(mappedBy = "goods", fetch = FetchType.LAZY)
//    List<Worker> workers;
}

File: Front/neo-store-app/src/assets/scripts/models/good.ts
export class Good {
  id: number;
  title: string;
  category: string;
  price: number;
  discount: number;
  description: string;
  imageURL: string;
  amount: number;

  constructor(
    id: number,
    title: string,
    category: string,
    price: number,
    discount: number,
    description: string,
    imageURL: string,
    amount: number,
  ){
    this.id = id;
    this.title = title;
    this.category = category;
    this.price = price;
    this.discount = discount;
    this.description = description;
    this.imageURL = imageURL;
    this.amount = amount;
  }
}